src/main/java/com/google/devtools/build/lib/query2/cquery/GraphOutputFormatterCallback.java

```diff
@@ -50,9 +51,20 @@ public String getName() {
               // Order graph output first by target label, then by configuration hash.
               Label label1 = ct1.getLabel();
               Label label2 = ct2.getLabel();
-              return label1.equals(label2)
-                  ? ct1.getConfigurationChecksum().compareTo(ct2.getConfigurationChecksum())
-                  : label1.compareTo(label2);
+              if (!label1.equals(label2)) {
+                return label1.compareTo(label2);
+              }
+              String checksum1 = ct1.getConfigurationChecksum();
+              String checksum2 = ct2.getConfigurationChecksum();
+              if (Objects.equals(checksum1, checksum2)) {
```

Comment:
This is the same as the compareTo below, and the checksum2 null check isn't needed, either. I'd simplify this as:

if (checksum1 == null) {
  return -1;
} else {
  return checksum1.compareTo(checksum2);
}

Reply:
How's this? I left the double null check because of this comment from https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html:

> Note that null is not an instance of any class, and e.compareTo(null) should throw a NullPointerException even though e.equals(null) returns false.

Reply:
Or am I missing something about the null check?

Reply:
Ah, I hadn't realized that. Okay, the second null check is necessary, then.